src/extensions/mvs/load-generic.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2023-2024 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Adam Midlik <midlik@gmail.com>
+ */
+
+import { StructureRepresentation3D } from '../../mol-plugin-state/transforms/representation';
+import { PluginContext } from '../../mol-plugin/context';
+import { StateBuilder, StateObject, StateObjectSelector, StateTransform, StateTransformer } from '../../mol-state';
+import { stringHash } from './helpers/utils';
+import { Kind, Subtree, SubtreeOfKind, Tree } from './tree/generic/tree-schema';
+import { dfs } from './tree/generic/tree-utils';
+
+
+/** Function responsible for loading a tree node `node` into Mol*.
+ * Should apply changes within `updateParent.update` but not commit them.
+ * Should modify `context` accordingly, if it is needed for loading other nodes later.
+ * `updateParent.selector` is the result of loading the node's parent into Mol* state hierarchy (or the hierarchy root in case of root node). */
+export type LoadingAction<TNode extends Tree, TContext> = (updateParent: UpdateTarget, node: TNode, context: TContext) => UpdateTarget | undefined
+
+/** Loading actions for loading a tree into Mol*, per node kind. */
+export type LoadingActions<TTree extends Tree, TContext> = { [kind in Kind<Subtree<TTree>>]?: LoadingAction<SubtreeOfKind<TTree, kind>, TContext> }
+
+/** Type for defining custom behavior when loading trees, usually based on node additional properties. */
+export interface LoadingExtension<TTree extends Tree, TContext, TExtensionContext> {
+    id: string,
+    description: string,
+    /** Runs before the tree is loaded */
+    createExtensionContext: (tree: TTree, context: TContext) => TExtensionContext,
```

Comment:
Should (could) this support async? Perhaps some extensions will require loading external resources.

```suggestion
    createExtensionContext: (tree: TTree, context: TContext) => TExtensionContext | Promise<TExtensionContext>,
```